make/test.mk

```diff
@@ -1,9 +1,117 @@
-COV_DIR = coverage
-GOPKGS = $(shell go list ./...)
+COV_DIR = $(OUT_DIR)/coverage
 
 .PHONY: test
 ## runs all tests with bundles assets
 test: generate
+	@echo "running the tests without coverage..."
+	# TODO enable -race flag, there are some DATA RACE need to be fixed
+	go test ${V_FLAG} -failfast ./...
+
+.PHONY: test-with-coverage
+## runs the tests with coverage
+test-with-coverage: generate
+	@echo "running the tests with coverage..."
 	@-mkdir -p $(COV_DIR)
-	@-rm -f $(COV_DIR)/coverage.txt
-	go test -timeout 10m -coverprofile=coverage.txt -covermode=atomic -coverpkg=$$GOPKGS ./...
+	@-rm $(COV_DIR)/coverage.txt
+	go test -timeout 10m -vet off ${V_FLAG} -coverprofile=$(COV_DIR)/coverage.txt -covermode=atomic ./...
+
+.PHONY: upload-codecov-report
```

Comment:
Let's update travis too so it doesn't publish coverage reports anymore - https://github.com/codeready-toolchain/registration-service/blob/master/.travis.yml#L43

Reply:
Done https://github.com/codeready-toolchain/registration-service/pull/51/commits/06c64dc9ec0cc87be3fd389cdc86f24c994b0588